qiskit/algorithms/amplitude_amplifiers/amplitude_amplifier.py

```diff
@@ -43,9 +43,9 @@ class AmplitudeAmplifierResult(AlgorithmResult):
 
     def __init__(self) -> None:
         super().__init__()
-        self._top_measurement = None
+        self._top_measurement: Optional[str] = None
```

Comment:
A newer syntax for this, `str | None` is the one preferred, since Python 3.10. But you can already enable this for older Python releases with `from __future__ import annotations`.